renderdoc/driver/vulkan/vk_core.cpp

```diff
@@ -6034,6 +6046,73 @@ void WrappedVulkan::UpdateImageStates(const rdcflatmap<ResourceId, ImageState> &
   }
 }
 
+size_t WrappedVulkan::CalculateImageMemoryCopyHostMemorySize(VkImage image, uint32_t memoryRowLength,
+                                                             uint32_t memoryImageHeight,
+                                                             const VkExtent3D &imageExtent)
+{
+  // VK_EXT_host_image_copy supports copying preswizzled data to/from images with the
+  // VK_HOST_IMAGE_COPY_MEMCPY flag. In that case, the contents of memoryRowLength and
+  // memoryImageHeight are meaningless, and the entirety of the image subresource is copied.
+  // However, the driver needs to be queried with vkGetImageSubresourceLayout2EXT to determine the
+  // amount of memory needed to hold the preswizzled image data.
+  //
+  // However, this flag will reduce the portability of the RenderDoc trace, and is removed from
+  // copies to and from host memory. The data is thus always assumed linear.
+
+  // If either of memoryRowLength and memoryImageHeight are 0, they are implicitly derived from
+  // imageExtent.
+  if(memoryRowLength == 0)
+  {
+    memoryRowLength = imageExtent.width;
+  }
+  if(memoryImageHeight == 0)
+  {
+    memoryImageHeight = imageExtent.height;
+  }
+
+  RDCASSERT(IsCaptureMode(m_State));
+
+  const auto &imageInfo =
+      GetResourceManager()->GetResourceRecord(GetResID(image))->resInfo->imageInfo;
+  const VkFormat format = imageInfo.format;
+
+  return GetByteSize(memoryRowLength, memoryImageHeight, imageExtent.depth, format, 0);
+}
+
+void WrappedVulkan::AdjustSubresourceHostMemcpySizeQuery(VkImage image,
```

Comment:
If the MEMCPY flag is not removed by RenderDoc, this function becomes unnecessary; it's adjusting the query to get the memcpy size. Instead, the function above it would have to make a call to `vkGetImageSubresourceLayout2` to figure out the host memory size instead of using `GetByteSize`.

I infinitely regret not pairing `pHostPointer` with a size. :(